home-work-app/src/app/types/MovieAction.ts

```diff
@@ -0,0 +1,14 @@
+import { IApiResponse } from '../interfaces/IApiResponse';
+
+export type MovieAction = {
```

Comment:
It is wrong.
You need to create a separate `Action` type for each action, and then combine them in reducer.
```
type Action = FilterMoviesAction | SortMoviesAction | SaveMoviesAction.
export const movieEditor: MovieActionReducer = (
    state: MovieState = initialState,
    action: Action,
): MovieState => {...}
```